packages/kong-plugin/kong/plugins/readme-plugin/handler.lua

```diff
@@ -0,0 +1,244 @@
+
+  return parsed_url
+end
+
+-- Creates a payload for ReadMe
+-- @return JSON string in ReadMe API Log format
+local function make_readme_payload(conf, entries)
+  local payload = {}
+  for _, entry in ipairs(entries) do
+    local request_entry = {
+      pageref = entry.request.url,
+      startedDateTime = os.date("!%Y-%m-%dT%H:%M:%SZ", entry.started_at / 1000),
+      time = entry.latencies.request,
+      request = {
+        httpVersion = entry.request.httpVersion,
+        headers = {},
+        method = entry.request.method,
+        queryString = {},
+        bodySize = entry.request.size,
+        url = entry.request.url,
+      },
+      response = {
+        status = entry.response.status,
+        headers = {},
+        content = {
+          mimeType = entry.response.mimeType,
+          size = entry.response.size,
+        },
+        bodySize = entry.response.size
+      }
+    }
+
+
+    -- Convert headers
+    for name, value in pairs(entry.request.headers) do
+      local final_value = value
+      if conf.hide_headers[name] then
+        if conf.hide_headers[name] == "" then
+          kong.log.debug("Hiding request header: ", name)
+          break
+        else
+          final_value = conf.hide_headers[name]
+          kong.log.debug("Overriding request header: ", name, " with ", final_value)
+        end
+      end
+      table.insert(request_entry.request.headers, {name = name, value = final_value})
+    end
+
+    for name, value in pairs(entry.response.headers) do
+      local final_value = value
+      if conf.hide_headers[name] then
+        if conf.hide_headers[name] == "" then
+          kong.log.debug("Hiding response header: ", name)
+          break
+        else
+          final_value = conf.hide_headers[name]
+          kong.log.debug("Overriding response header: ", name, " with ", final_value)
+        end
+      end
+      table.insert(request_entry.response.headers, {name = name, value = final_value})
+    end
+
+    for name, value in pairs(entry.request.querystring) do
+      table.insert(request_entry.request.queryString, {name = name, value = value})
+    end
+
+    local group = {}
+    if conf.group_by then
+      for header_name, group_value in pairs(conf.group_by) do
+        local header_value = entry.request.headers[header_name]
+        group[group_value] = header_value or "unknown"
+      end
+    end
+
+    local id_header = (conf.id_header and conf.id_header:lower()) or "authorization"
+    local id = entry.request.headers[id_header] or "unknown"
+    if id_header == "authorization" and entry.raw_auth then
+      id = entry.raw_auth
+    end
+    group.id = id
+    local api_log = {
+      httpVersion = entry.httpVersion,
+      requestBody = {},
+      responseBody = {},
+
+      clientIPAddress = entry.client_ip,
+      createdAt = request_entry.startedDateTime,
+      development = false,
+      error = nil,
+      group = group,
+      _id = entry.request.id,
```

Comment:
This is typically a UUID that we create in the client. I'm not seeing that done here, am I missing something?

Reply:
kong creates unique id, thats where this `id` is coming from